examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -707,46 +699,48 @@ void ProcessEvent(WfxEvent_t inEvent)
     case WFX_EVT_SCAN:
 #ifdef SL_WFX_CONFIG_SCAN
         rsi_rsp_scan_t scan_rsp = { 0 };
-        int32_t status          = rsi_wlan_bgscan_profile(1, &scan_rsp, sizeof(scan_rsp));
+        memset(&scan_rsp, 0, sizeof(scan_rsp));
+        int32_t status = rsi_wlan_bgscan_profile(1, &scan_rsp, sizeof(scan_rsp));
 
-        if (status)
+        if (status != RSI_SUCCESS)
         {
-            SILABS_LOG("SSID scan failed: %02x ", status);
+            SILABS_LOG("rsi_wlan_bgscan failed: %02x ", status);
+            return;
         }
-        else
+        rsi_scan_info_t * scan;
+        wfx_wifi_scan_result_t ap;
+        for (int x = 0; x < scan_rsp.scan_count[0]; x++)
         {
-            rsi_scan_info_t * scan;
-            wfx_wifi_scan_result_t ap;
-            for (int x = 0; x < scan_rsp.scan_count[0]; x++)
+            scan = &scan_rsp.scan_info[x];
+            // is it a scan all or target scan
+            if (!wfx_rsi.scan_ssid || (wfx_rsi.scan_ssid && strcmp(wfx_rsi.scan_ssid, (char *) scan->ssid) == CMP_SUCCESS))
```

Comment:
strcmp can overrun, please sure all lengths/termination are validated before processing.

Overall, it would be MUCH BETTER to avoid null-termianted string at all in Wi-Fi driver... You don't want edge conditions where MAX_SSID_LENGTH happens, there is no null-termination and you have some overrun from the air.

This is really hard to whack-a-mole and is better done in a holistic "let's avoid null-termination" methods.

If null-terminated strings absolutely have to be used, it is required to follow *strict* hygiene and never use non-`n` variants e.g. `strcmp` instead of `strncmp`, and since you are not using `strlcpy` (which is not ANSI C), you need to force-null-terminate every time `strncpy` happens unless you had done a memset first (which is hard to audit).